src/plugins/intel_npu/src/plugin/npuw/partitioning/partitioning.hpp

```diff
@@ -73,15 +72,24 @@ struct Subgraph {
     QuantUnpackGather _quant_unpack_gather;
 
     using Ref = std::reference_wrapper<Subgraph>;
+
+    void settag(const std::string& t) {
+        LOG_DEBUG("Subgraph set-tag=" << t);
+        _tag = t;
+    }
+    std::string gettag() const {
+        return _tag;
+    }
```

Comment:
If you want to stick to the controlled API I'd propose the following change:
```suggestion
    void tag(const std::string& t) {
        // Sets the tag
     }
    std::string tag() const {
        // Gets the tag
    }
```
It should look more concise in the user code. Also, as there's three cases where the same behavior is implemented, it may make sense to introduce a `Taggable` trait with the above methods and derive our three structures from that.